pyqtgraph/graphicsItems/BarGraphItem.py

```diff
@@ -178,4 +212,48 @@ def name(self):
         return self.opts.get('name', None)
 
     def getData(self):
-        return self.opts.get('x'),  self.opts.get('height')
+        return self.x0, self.width, self.y0, self.height
+
+    def dataBounds(self, ax, frac=1.0, orthoRange=None):
```

Comment:
This implementation doesn't look quite right. For example, in the `ax == 0` case, the bar width isn't taken into account.

I think both `ax == 0` and `ax == 1` cases should look basically identical with x/y swapped, and both need to take into account horizontal and vertical bar orientations.

Reply:
Didn't even consider this; but this is right, BarGraphItem should be able to handle horizontal bars or vertical bars.